lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -366,6 +366,27 @@ sub register ($self, $app, $config) {
 
             $c->res->headers->links($links);
         });
+
+    $app->helper(
+        regex_problem => sub ($c, $regexes, $context = undef) {
+            my $regex_problem;
+            for my $regex_string (@$regexes) {
+                # treat regex warnings as fatal as apparently not all problems are fatal errors
+                # note: We should not leave those warnings unhandled as they would end up in the log.
+                #       An example for such a warning is "$* matches null string many times in regex".
+                use warnings FATAL => 'regexp';
+                # test regex compilation
+                my $parsed_regex = eval { qr/$regex_string/ };
+                if ($@) { $regex_problem = $@ }
+                # test regex matching as some problems are only warned about when matching
+                elsif ($parsed_regex) { '' =~ $parsed_regex }
```

Comment:
Can you give an example where the warning only appears when matching?
And what would that code here then do? with the fatal warnings, it would die, but it's not in the eval block anymore...

Reply:
It's probably not possible to catch all possible warnings with having a match on the empty string.
So I just looked through [perldoc perldiag](https://perldoc.perl.org/perldiag), lloking for `(W regexp)`, and I think warnings like `Complex regular subexpression recursion limit (%d) exceeded` would only appear in the real matching.

Doing the eval at the actual match would make the code a bit more complicated, I guess.

Reply:
> Doing the eval at the actual match would make the code a bit more complicated, I guess.

Not sure how to solve this. Maybe this validation is really just good enough as a first sanity check. So the places where we actually use the regex would use some "safe" helper function to do the matching with proper error handling.

Reply:
The version I've just pushed fixes all problems. So `'' =~ $parsed_regex` has been moved inside the `eval`-block and regex errors/warnings are also handled when doing actual matching later on.